TFNetworkRecLayer.py

```diff
@@ -943,30 +942,25 @@ def _construct(self, prev_outputs, prev_extra, i,
     :param dict[str,tf.Tensor] prev_outputs: outputs of the layers from the previous step
     :param dict[str,dict[str,tf.Tensor]] prev_extra: extra output / hidden states of the previous step for layers
     :param tf.Tensor i: loop counter. scalar, int32, current step (time)
-    :param tf.Tensor|None data: optional source data, shape e.g. (batch,dim)
-    :param tf.Tensor|None classes: optional target classes, shape e.g. (batch,) if it is sparse
+    :param dict[str,tf.Tensor|None] data: All data needed from outside of the loop. Usually includes
+        'source' data (shape e.g. (batch,dim)) and target classes (shape e.g. (batch,) if it is sparse).
```

Comment:
Target is `'target'`? Or `'classes'`? Be explicit.

Reply:
Why do you allow `None` as value?

Reply:
None was also allowed before, it means e.g. that the recurrent unit has no external input data. But I guess you mean I should not use None but check if the key is in the dict instead, right? 
For other data than "source" the key is the one used in `parent_net.extern_data` (see line 763), but I can add that in the description.

Reply:
`None` was there because `data` has to have some value because it exists always (the parameter). But if it is a dict, then `data["data"]` does not have to exists. I don't understand what the point is to have `data["data"] is None`.

The new description is also not really clear. `classes` is supposed to be a key? Then write `'classes'` to make that clear.